mungegithub/mungers/publisher_issue.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/golang/glog"
+
+	githubapi "github.com/google/go-github/github"
+	"k8s.io/test-infra/mungegithub/github"
+)
+
+const Title = "Repo publisher failure"
+
+// TODO: before merge, change this to @kubernetes-repo-publishing-cops, which will start with @caesarxuchao @sttts @deads2k @lavalamp
+const owners = "@caesarxuchao"
+
+// publisherFailure includes the information to create an issue
+type publisherFailure struct {
+	err error
+	log string
+}
+
+func (p *publisherFailure) title() string {
+	return Title
+}
+
+func (p *publisherFailure) body() string {
+	body := p.err.Error()
+	body += fmt.Sprintf("\n")
+	body += fmt.Sprintf("cc %s", owners)
+	body += fmt.Sprintf("\n")
+	body += p.log
+	return body
+}
+
+// publisherIssueTracker tracks latest open issue regarding publisher failure,
+// can update the existing issue or create a new one.
+type publisherIssueTracker struct {
+	config             *github.Config
+	lastestOpenIssue   *githubapi.Issue
+	lastestIssueNumber int
+}
+
+func NewPublisherIssueTracker(config *github.Config) *publisherIssueTracker {
+	return &publisherIssueTracker{config: config}
+}
+
+func (p *publisherIssueTracker) Init() error {
+	issues, err := p.config.ListAllIssues(&githubapi.IssueListByRepoOptions{
```

Comment:
test-infra currently has 136 open issues, shouldn't take long.

For example, the test flake robot lists 436 [issues](https://github.com/kubernetes/kubernetes/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aopen%20label%3Akind%2Fflake%20)